### PR TITLE
Migration Guide for Deprecations removed since last stable release (1.22)

### PR DESCRIPTION
**Pull request by Piinks**
This migration guide covers all of the deprecations that have reached end of life and been removed since the last stable release (1.22).

The aim is to provide each of the deprecations that have been removed with a simple guide for migration and reference. Most deprecations already have a guide as they are considered breaking, but most of these older ones pre-date that policy and so do not already have a migration guide.

Rather than create a  guide for each individual change, this seemed like a cleaner approach as they are all changing for the same reason.

In future versions, we could just reference the existing guide that was created at time of deprecation to aid in discovery.

### CODE REVIEW

**Comment by Piinks** [comment]
(Post-holidays) @Sfshaza2 & @Hixie Do you have any opinions on formatting here? I have just pushed a template to review before filling in all of the deprecations slated for removal for the next release.

The full list is in https://flutter.dev/go/deprecation-lifetime :)

**Comment by Piinks** [comment]
@sfshaza2 & @kwalrath re upcoming freeze, I'm still looking for some feedback here. We removed a lot of deprecations recently and I wanted to put together a hybrid-breaking change doc that covers them all, just need to sort out the template first.

They are already listed in flutter.dev/go/deprecations-removed-after-1-22 do you think that covers it? Or should I write out a more fully-fledged doc here?

**Comment by sfshaza2** [comment]
I've added csells@ to this thread, as this list would be of interest to him.

It seems pretty clear to me what this doc means, so I don't feel like you
need more explanation. But rather than say "we removed deprecations", use
"we removed deprecated APIs". Otherwise, it confuses me and sounds like
maybe you rescinded those deprecations.

**Comment by sfshaza2** [comment]
+1 to "we removed deprecated APIs" instead of "we removed deprecations" for
the reason Shams mentioned. Thanks, Kate!

**Comment by Piinks** [comment]
@sfshaza2 & @kwalrath This is at about the halfway point, since it is so large I thought I'd check in and confirm the format I'm using for this looks ok before I write the rest. :)
The quick reference to see all of the ones that will be listed in this format is here: flutter.dev/go/deprecations-removed-after-1-22

**Review by lucalves — APPROVED** [review]

**Comment by Piinks** [comment]
I haven't gotten feedback that the format is ok, but this is time sensitive so I went ahead and filled in the rest.
 This now has all of the content that is intended to be included in this guide and is ready for review.

File: src/docs/release/breaking-changes/1-22-deprecations.md [hunk 1]
```diff
@@ -0,0 +1,768 @@
...
+[#73748]: {{site.github}}/flutter/flutter/pull/73748
+
+---
+
+### `Layer.findAll`
+
+Supported by fix tool: No
+
+The `Layer.findAll` method was deprecated with the introduction of `Layer.findAnnotations` in order
+to unify the implementations of `find` and `findAll`. To migrate affected code, call on
```
Comment (justinmc) [inline src/docs/release/breaking-changes/1-22-deprecations.md:505]: Nit: "call on" => "call"

If I'm reading this correctly.

File: src/docs/release/breaking-changes/1-22-deprecations.md [hunk 2]
```diff
@@ -0,0 +1,768 @@
...
+---
+
+### Generic methods for `BuildContext`
+
+Supported by fix tool: Yes
+
+Several methods in `BuildContext` were using `Type` to search for ancestors. Most of those methods
+implied a cast at call site because their return type was a parent type. Moreover the type provided
+was not checked at analysis time even if the type is actually constrained. Making these methods
+generics improves type safety and need less code.
```
Comment (justinmc) [inline src/docs/release/breaking-changes/1-22-deprecations.md:605]: Nit: "need less code" => "requires less code"

Or maybe these were copied from somewhere and shouldn't change.

File: src/docs/release/breaking-changes/1-22-deprecations.md [hunk 3]
```diff
@@ -0,0 +1,768 @@
...
+* Removed in [#73754][]
+ 
+[`WaitForCondition`]: {{site.api}}/flutter/flutter_driver/WaitForCondition-class.html
+[#37736]: {{site.github}}/flutter/flutter/pull/37736
+[#38836]: {{site.github}}/flutter/flutter/pull/38836
+[#73754]: {{site.github}}/flutter/flutter/pull/73754
+
+---
+
+
```
Comment (justinmc) [inline src/docs/release/breaking-changes/1-22-deprecations.md:765]: Nit: Too much whitespace?

**Review by justinmc — APPROVED** [review]
LGTM from my point of view 👍

File: src/docs/release/breaking-changes/1-22-deprecations.md [hunk 4]
```diff
@@ -0,0 +1,768 @@
+---
+title: Deprecated API removed after v1.22
+description: After reaching end of life, the following deprecated APIs were removed from Flutter.
+---
+
+## Summary
+
+In accordance with Flutter's [Deprecation Policy][], deprecated APIs that reached end of life after the
+1.22 stable release have been removed. This is the first time deprecated APIs have been removed from
+Flutter, and some of these deprecations pre-date our migration guide policy.
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/1-22-deprecations.md:10]:
```suggestion
Flutter, and some of these deprecations predate our migration guide policy.
```

File: src/docs/release/breaking-changes/1-22-deprecations.md [hunk 5]
```diff
@@ -0,0 +1,768 @@
...
+
+All affected APIs have been compiled into this primary source to aid in migration. A
+[quick reference sheet] is available as well.
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/1-22-deprecations.md:13]:
```suggestion
[quick reference sheet][] is available as well.
```

File: src/docs/release/breaking-changes/1-22-deprecations.md [hunk 6]
```diff
@@ -0,0 +1,768 @@
...
+
+A [design document] and [article] are available for more context on Flutter's deprecation policy.
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/1-22-deprecations.md:15]:
```suggestion
A [design document][] and [article][] are available for more context on Flutter's deprecation policy.
```

File: src/docs/release/breaking-changes/1-22-deprecations.md [hunk 7]
```diff
@@ -0,0 +1,768 @@
...
+
+[Deprecation Policy]: {{site.github}}/flutter/flutter/wiki/Tree-hygiene#deprecation
+[quick reference sheet]: /go/deprecations-removed-after-1-22
+[design document]: /go/flutter-deprecation-lifetime
+[article]: https://medium.com/flutter/deprecation-lifetime-in-flutter-e4d76ee738ad
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/1-22-deprecations.md:20]:
```suggestion
[article]: {{site.flutter-medium}}/deprecation-lifetime-in-flutter-e4d76ee738ad
```

File: src/docs/release/breaking-changes/1-22-deprecations.md [hunk 8]
```diff
@@ -0,0 +1,768 @@
...
+
+## Changes
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/1-22-deprecations.md:22]: We need to have at least one sentence of text between headings. How about:

```
This section lists the deprecations, listed by the affected class.
```

File: src/docs/release/breaking-changes/1-22-deprecations.md [hunk 9]
```diff
@@ -0,0 +1,768 @@
...
+
+### Pointer Events Constructed `fromHoverEvent`
+
+Supported by fix tool: Yes
+ 
+The `fromHoverEvent` constructors for `PointerEnterEvent` and `PointerExitEvent` were deprecated
+in v1.4.3.
+The `fromMouseEvent` constructor should be used instead.
+ 
+#### Migration Guide
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/1-22-deprecations.md:194]: => #### Migration guide

(We use sentence case headers.)

File: src/docs/release/breaking-changes/1-22-deprecations.md [hunk 10]
```diff
@@ -0,0 +1,768 @@
...
+---
+
+### `showDialog` uses `builder`
+
+Supported by fix tool: Yes
+ 
+The `child` parameter of `showDialog` was deprecated in in v0.2.3.
+The `builder` parameter should be used instead.
+ 
+#### Migration Guide
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/1-22-deprecations.md:238]: => Migration guide

Global change.

File: src/docs/release/breaking-changes/1-22-deprecations.md (hunk 1 above)
Comment (sfshaza2) [reply src/docs/release/breaking-changes/1-22-deprecations.md:505]: +1

File: src/docs/release/breaking-changes/1-22-deprecations.md (hunk 2 above)
Comment (sfshaza2) [reply src/docs/release/breaking-changes/1-22-deprecations.md:605]: +1

File: src/docs/release/breaking-changes/1-22-deprecations.md [hunk 11]
```diff
@@ -0,0 +1,768 @@
+---
+title: Deprecated API removed after v1.22
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/1-22-deprecations.md:2]: Looking at the generated TOC, I don't think we need so many levels of headers. Maybe just h1 and h2. The deeper level of headers can be converted to bolded text or similar.

**Review by sfshaza2 — CHANGES_REQUESTED** [review]

File: src/docs/release/breaking-changes/1-22-deprecations.md (hunk 11 above)
Comment (Piinks) [reply src/docs/release/breaking-changes/1-22-deprecations.md:2]: I reduced the number of headings to h1,2,3. The third heading is for the individual class/change, which makes each of the changes in these guides linkable and accessible.

**Review by Piinks — COMMENTED** [review]
Updated, PTAL. :)

**Review by sfshaza2 — APPROVED** [review]
LGTM

File: src/docs/release/breaking-changes/1-22-deprecations.md [hunk 12]
```diff
@@ -0,0 +1,772 @@
...
+Flutter, and some of these deprecations predate our migration guide policy.
+
+All affected APIs have been compiled into this primary source to aid in migration. A
+[quick reference sheet][] is available as well.
+
+A [design document][] and [article][] are available for more context on Flutter's deprecation policy.
+
+[Deprecation Policy]: {{site.github}}/flutter/flutter/wiki/Tree-hygiene#deprecation
+[quick reference sheet]: /go/deprecations-removed-after-1-22
+[design document]: /go/flutter-deprecation-lifetime
```
Comment (sfshaza2) [inline src/docs/release/breaking-changes/1-22-deprecations.md:19]: Kate, this link doesn't work and has broken the build. Please fix...

Comment (Piinks) [reply src/docs/release/breaking-changes/1-22-deprecations.md:19]:
```suggestion
[design document]: /go/deprecation-lifetime
```

**Pull request merged**